src/main/java/org/opensearch/ad/transport/ADJobRunnerTransportAction.java

```diff
@@ -0,0 +1,123 @@
+import org.opensearch.sdk.SDKClient.SDKRestClient;
+import org.opensearch.tasks.Task;
+import org.opensearch.transport.TransportService;
+
+/**
+ * Transport Action to execute job runner of anomaly detection.
+ */
+public class ADJobRunnerTransportAction extends HandledTransportAction<ExtensionActionRequest, ExtensionActionResponse> {
+
+    private static final Logger LOG = LogManager.getLogger(ADJobRunnerTransportAction.class);
+
+    private SDKRestClient client;
+
+    private AnomalyDetectorJob scheduledJobParameter;
+
+    private final NamedXContentRegistry namedXContentRegistry;
+
+    protected ADJobRunnerTransportAction(
+        TransportService transportService,
+        ActionFilters actionFilters,
+        SDKRestClient client,
+        ExtensionsRunner extensionsRunner
+    ) {
+        super(ADJobRunnerAction.NAME, transportService, actionFilters, ExtensionActionRequest::new);
+        this.client = client;
+        this.namedXContentRegistry = extensionsRunner.getNamedXContentRegistry().getRegistry();
+    }
+
+    @Override
+    protected void doExecute(Task task, ExtensionActionRequest request, ActionListener<ExtensionActionResponse> actionListener) {
+        String errorMessage = "Failed to run the Job";
+        ActionListener<ExtensionActionResponse> listener = wrapRestActionListener(actionListener, errorMessage);
+        JobRunnerRequest jobRunnerRequest;
+        try {
+            jobRunnerRequest = new JobRunnerRequest(request.getRequestBytes());
+            CompletableFuture<AnomalyDetectorJob> inProgressFuture = new CompletableFuture<>();
+            String jobParameterDocumentId = jobRunnerRequest.getJobParameterDocumentId();
+            if (jobParameterDocumentId == null || jobParameterDocumentId.isEmpty()) {
+                listener.onFailure(new IllegalArgumentException("jobParameterDocumentId cannot be empty or null"));
+            } else {
+                findById(jobParameterDocumentId, new ActionListener<>() {
+                    @Override
+                    public void onResponse(AnomalyDetectorJob anomalyDetectorJob) {
+                        scheduledJobParameter = anomalyDetectorJob;
+                        inProgressFuture.complete(scheduledJobParameter);
+                    }
+
+                    @Override
+                    public void onFailure(Exception e) {
+                        logger.info("could not find AnomalyDetectorJob with id " + jobParameterDocumentId, e);
+                        inProgressFuture.completeExceptionally(e);
+                    }
+                });
+                JobExecutionContext jobExecutionContext = jobRunnerRequest.getJobExecutionContext();
+
+                AnomalyDetectorJobRunner.getJobRunnerInstance().runJob(scheduledJobParameter, jobExecutionContext);
+                JobRunnerResponse jobRunnerResponse = new JobRunnerResponse(true);
+                listener.onResponse(new ExtensionJobActionResponse<>(jobRunnerResponse));
```

Comment:
The proxy `ScheduledJobRunner` [here](https://github.com/opensearch-project/job-scheduler/blob/27aace4785ef22d7710ff1a2fdabbf4748941dd4/src/main/java/org/opensearch/jobscheduler/utils/JobDetailsService.java#L251) has a timeout value of 30 seconds to wait for a response from an extension regarding the status of their job runner execution. 

I am concerned with this implementation since we're running the job prior to sending back a response. The length of a job can vary (this is indicated by the lock duration seconds required for this job), which means that the job execution time may be longer than the response timeout.

Here we are pulling the `ScheduledJobParameter` from the job index and extracting the `JobExecutionContext` from the request, I propose that we validate the `JobExecutionContext` (ie validating that the params are not null or empty) and also validate that we're able to successfully able retrieve the `ScheduledJobParameter` from the extension's job index, and return a response based on this validation. Once the response is returned, we can execute the job. Since the onus of job execution is on the extension, there's no need for job scheduler to wait for the execution to finish to receive a response

Reply:
ACK working on it